modules/icgc-argo-workflows/checkinput/resources/usr/bin/dnaaln.py

```diff
@@ -0,0 +1,377 @@
+            raise AssertionError("'patient' input is required.")
+
+    def _validate_sex(self, row):
+        """Assert that expected sex is correct."""
+        if len(row[self._sex_col]) <= 0:
+            raise AssertionError("'analysis_type' input is required.")
+        if row[self._sex_col]!="XX" and row[self._sex_col]!="XY" and row[self._sex_col]!="NA":
+            raise AssertionError("sex should be one of the following values : XX,XY,NA")
+
+    def _validate_status(self, row):
+        """Assert that expected tumour status is correct."""
+        if len(row[self._status_col]) <= 0:
+            raise AssertionError("'status' input is required.")
+        if row[self._status_col]!="1" and row[self._status_col]!="0":
+            raise AssertionError("Tumour status should be \"0\" is normal else \"1\"")
+
+    def _validate_sample(self, row):
+        """Assert that expected sample is correct."""
+        if len(row[self._sample_col]) <= 0:
+            raise AssertionError("'sample' input is required.")
+    
+
+    def _validate_lane(self, row):
+        """Assert that expected lane is correct."""
+        if len(row[self._lane_col]) <= 0:
+            raise AssertionError("'lane' input is required.")
+    
+
+    def _validate_fastq_1(self, row):
+        """Assert that expected fastq_1 is correct."""
+        if len(row[self._fastq_1_col]) <= 0:
+            raise AssertionError("'fastq_1' input is required.")
+        if not (
+            row[self._fastq_1_col].endswith(".fq.gz") or 
+            row[self._fastq_1_col].endswith(".fastq.gz") or
+            row[self._fastq_1_col].endswith(".bam")
+            ):
+            raise AssertionError("'fastq_1' incorrect format detected.")
+    
+
+    def _validate_fastq_2(self, row):
+        """Assert that expected fastq_2 is correct."""
+        if row[self._single_end_col]=="True":
+            return 
+
+        if len(row[self._fastq_2_col]) <= 0:
+            raise AssertionError("'fastq_2' input is required.")
+        if row[self._fastq_2_col].endswith(".fastq.gz"):
+            if row[self._fastq_2_col].split("/")[-1].replace("R2.fastq.gz","")!=row[self._fastq_1_col].split("/")[-1].replace("R1.fastq.gz",""):
+                raise AssertionError("'fastq_1' and 'fastq_2' prefix differ.")
+        if row[self._fastq_2_col].endswith(".fq.gz"):
+            if row[self._fastq_2_col].split("/")[-1].replace("R2.fq.gz","")!=row[self._fastq_1_col].split("/")[-1].replace("R1.fq.gz",""):
+                raise AssertionError("'fastq_1' and 'fastq_2' prefix differ.")
+        if row[self._fastq_2_col].endswith(".bam"):
+            if row[self._fastq_2_col]!=row[self._fastq_1_col]:
+                raise AssertionError("'fastq_1' and 'fastq_2' prefix differ.")
+
+    def _validate_read_group(self, row):
+        """Assert that expected read_group is correct."""
+        if len(row[self._read_group_col]) <= 0:
+            raise AssertionError("'read_group' input is required.")
```

Comment:
DNA alignment workflow needs the read_group info, which can be compiled from the provided metatada.

Reply:
@edsu7 requesting `read_group` string directly from `csv` field is not a use friendly way. 
How about we add the following columns into csv fields:
- library_name (required)
- platform_unit (required)
- platform (optional)
- sequencing_center (optional)
- sequencing_date (optional)
- platform_model (optional)
then we compile the `read_group` string on the fly based on these fields.

Reply:
Added the following:
https://github.com/icgc-argo-workflows/argo-modules/blob/e9ee3bc1a463e0d7d20e410246c0f016cff06d49/modules/icgc-argo-workflows/checkinput/resources/usr/bin/dnaaln.py#L140-L175